Samples/excel-add-in-custom-functions-using-web-workers/src/functions/functions.js

```diff
@@ -0,0 +1,192 @@
+﻿/* global clearInterval, console, setInterval */
+
+/**
+ * Add two numbers.
+ * @customfunction
+ * @param {number} first First number.
+ * @param {number} second Second number.
+ * @returns {number} The sum of the two numbers.
+ */
+export function add(first, second) {
+  return first + second;
+}
+
+/**
+ * Displays the current time once a second.
+ * @customfunction
+ * @param {CustomFunctions.StreamingInvocation<string>} invocation Custom function invocation.
+ */
+export function clock(invocation) {
+  const timer = setInterval(() => {
+    const time = currentTime();
+    invocation.setResult(time);
+  }, 1000);
+
+  invocation.onCanceled = () => {
+    clearInterval(timer);
+  };
+}
+
+/**
+ * Returns the current time.
+ * @returns {string} String with the current time formatted for the current locale.
+ */
+export function currentTime() {
+  return new Date().toLocaleTimeString();
+}
+
+/**
+ * Increments a value once a second.
+ * @customfunction
+ * @param {number} incrementBy Amount to increment.
+ * @param {CustomFunctions.StreamingInvocation<number>} invocation
+ */
+export function increment(incrementBy, invocation) {
+  let result = 0;
+  const timer = setInterval(() => {
+    result += incrementBy;
+    invocation.setResult(result);
+  }, 1000);
+
+  invocation.onCanceled = () => {
+    clearInterval(timer);
+  };
+}
+
+/**
+ * Writes a message to console.log().
+ * @customfunction LOG.
+ * @param {string} message String to write.
+ * @returns String to write.
+ */
+export function logMessage(message) {
+  console.log(message);
+
+  return message;
+}
+
+const SampleNamespace = {};
+
+(function (SampleNamespace) {
+  // The max number of web workers to be created.
+  const g_maxWebWorkers = 4;
+
+  // The array of web workers.
+  const g_webworkers = [];
+
+  // Next job id.
```

Comment:
When talking about an ID in a sentence or comment, use "ID" (short of identification). Lowercase "id" means something different (the impulsive part of the human personality, according to Freud - countered by the ego and superego). 
```suggestion
  // Next job ID.
```